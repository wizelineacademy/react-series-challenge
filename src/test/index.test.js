import { render } from 'react-dom';
import '../';
jest.mock('react-dom');

describe('index', () => {
  it('Should check if the application was render', () => {
    expect(render).toHaveBeenCalledTimes(1);
  });
});
