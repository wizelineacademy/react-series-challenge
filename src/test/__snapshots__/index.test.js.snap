// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the app correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <styled.nav>
                <Styled(NavLink)
                        to="/"
                >
                        Trending gifs
                </Styled(NavLink)>
                <Styled(NavLink)
                        to="/favorites"
                >
                        Mis favoritos
                </Styled(NavLink)>
        </styled.nav>
        <Switch>
                <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/favorites"
                />
                <Redirect
                        push={false}
                        to="/"
                />
        </Switch>
</div>,
      "theme": Object {
        "primary": Object {
          "background": "#000",
          "foreground": "#fff",
        },
        "secondary": Object {
          "background": "#4965f3",
          "foreground": "#fff",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.nav>
            <Styled(NavLink)
                        to="/"
            >
                        Trending gifs
            </Styled(NavLink)>
            <Styled(NavLink)
                        to="/favorites"
            >
                        Mis favoritos
            </Styled(NavLink)>
</styled.nav>,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/favorites"
            />
            <Redirect
                        push={false}
                        to="/"
            />
</Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(NavLink)
                to="/"
>
                Trending gifs
</Styled(NavLink)>,
              <Styled(NavLink)
                to="/favorites"
>
                Mis favoritos
</Styled(NavLink)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Trending gifs",
                "to": "/",
              },
              "ref": null,
              "rendered": "Trending gifs",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Mis favoritos",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Mis favoritos",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/favorites"
/>,
              <Redirect
                push={false}
                to="/"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "push": false,
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <styled.nav>
                    <Styled(NavLink)
                              to="/"
                    >
                              Trending gifs
                    </Styled(NavLink)>
                    <Styled(NavLink)
                              to="/favorites"
                    >
                              Mis favoritos
                    </Styled(NavLink)>
          </styled.nav>
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/favorites"
                    />
                    <Redirect
                              push={false}
                              to="/"
                    />
          </Switch>
</div>,
        "theme": Object {
          "primary": Object {
            "background": "#000",
            "foreground": "#fff",
          },
          "secondary": Object {
            "background": "#4965f3",
            "foreground": "#fff",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <styled.nav>
              <Styled(NavLink)
                            to="/"
              >
                            Trending gifs
              </Styled(NavLink)>
              <Styled(NavLink)
                            to="/favorites"
              >
                            Mis favoritos
              </Styled(NavLink)>
</styled.nav>,
            <Switch>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/favorites"
              />
              <Redirect
                            push={false}
                            to="/"
              />
</Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Styled(NavLink)
                  to="/"
>
                  Trending gifs
</Styled(NavLink)>,
                <Styled(NavLink)
                  to="/favorites"
>
                  Mis favoritos
</Styled(NavLink)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Trending gifs",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Trending gifs",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Mis favoritos",
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": "Mis favoritos",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/favorites"
/>,
                <Redirect
                  push={false}
                  to="/"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/favorites",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "push": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
